### DISCUSSION TITLE
Unable to fetch ServerConfig: Failed to decode access token: JWT must have 3 parts

### DISCUSSION THREAD

**Discussion by fritzfranzke** [dani-garcia/vaultwarden#4503] [Q&A]
### Subject of the issue
Bitwarden client on Linux fails to load password (infinite spinner) with the error `Unable to fetch ServerConfig: Failed to decode access token: JWT must have 3 parts` shown in the logs. Server logs show no errors. iOS client as well as Webvault works without issues. If this is solely Bitwarden client related feel free to close. 

### Deployment environment

* vaultwarden version: `1.30.5` 

* Install method: Docker Compose

* Clients used: Official Bitwarden Linux Client  `Version 2024.4.0`

* Reverse proxy and version: Caddy `v2.7.6`

* Other relevant details:

### Steps to reproduce
Deploy vaultwarden and official Bitwarden Linux client and try to log in.

### Expected behaviour
Login loads passwords

### Actual behaviour
Infinite spinner is shown in UI

**Comment by denysvitali** [comment]
I had a similar issue (while logged in).

My workaround was to logout and then log back in, after that, syncing works again and "Failed to decode access token: JWT must have 3 parts" doesn't appear anymore.

**Reply by fritzfranzke** [reply]
Hey, thanks for the quick reply! Unfortunately this didn't work for me, the same error persists.

**Reply by julianfoad** [reply]
Upstream issue:

- https://github.com/bitwarden/clients/issues/8722

The same underlying error can manifest both when trying to log in and when trying to add or edit an entry, if already logged in, when it manifests as a pop-up red error box saying "Failed to decode access token: JWT must have 3 parts".

For "snap" installations (e.g. on Ubuntu), TL;DR: follow their updated [post-install instructions](https://bitwarden.com/help/getting-started-desktop/#tab-3-6vQUhrVotSKFarA3cqyESG). That worked for me.

**Reply by fritzfranzke** [reply]
@julianfoad thanks for the heads-up, I'll keep an eye on it! For now I'm using the webvault as a workaround

**Reply by LeonardoRez** [reply]
same problem here. On Archlinux, 

Version 2024.4.0
Shell 28.3.1
Renderer 120.0.6099.291
Node 18.18.2
Architecture x64

I tried to log out and back in, but now I can't login. It prints "Error: The name is not activatable" and "Unable to fetch ServerConfig: Failed to decode access token: JWT must have 3 parts"

**Reply by BlackDex** [reply]
I do not seem to have any issues with the AppImage provided by Bitwarden.
And, the same for the Arch Linux Bitwarden Desktop client.

Check your reverse proxy configs and logs and also check the Vaultwarden logs.

**Reply by farzadha2** [reply]
yeah currently having the same issue

**Reply by farzadha2** [reply]
just reversed to version 2024.3.0  and it worked

**Reply by ghost** [reply]
The same problem on Archlinux

**Reply by ghost** [reply]
same problem here with ArchLinux

**Reply by rhoot** [reply]
I had this problem on Arch, and the issues linked below helped explain why: They made the Bitwarden client store tokens in whatever wallet you have. In my case, I had kwallet installed... but I wasn't using it, and my login manager was not configured to open it when I log in. So Bitwarden tried to use it but couldn't.

Fixing that still didn't seem to solve it, but deleting the Bitwarden config afterward did.

So my full fix was:

1. [Configure kwallet to unlock on login](https://wiki.archlinux.org/title/KDE_Wallet#Unlock_KDE_Wallet_automatically_on_login)
2. Log out, and back in of my DE
3. `rm -rf ~/.config/Bitwarden`
4. Launch and authenticate in Bitwarden again

A bit annoying that it requires using a secrets manager now, but it sounds like they're working on a fix to fall back to the old file-based behavior when one isn't available. And it probably also works with other secret managers, kwallet just happened to be what I picked.

Edit: The above is also why that snap command works for people using the snap; it gives the snap access to the secrets manager.

**Reply by mortbauer** [reply]
Thank you, that worked for me as well!!

**Reply by Blackilykat** [reply]
Had this issue on Arch Linux, fixed by downgrading to 2024.3.0.
My kwallet password was empty so I don't think that would've been the issue for me.

**Reply by rhoot** [reply]
You may just need to make sure kwallet runs on login then, if you're not using KDE. Not sure which WM/DE you're using but in sway I have `exec kwalletd5` in my config for that.

Edit: I guess that should be `exec kwalletd6` now.

**Reply by Blackilykat** [reply]
That worked, thanks! I thought having it open from kwallet manager would've been enough but I guess not.
This issue does seem to be [fixed](https://github.com/bitwarden/clients/pull/8913) but the new desktop version hasn't been released yet.

**Comment by BlackDex** [comment]
Seems a client issue 
- https://github.com/bitwarden/clients/issues/8651
- https://github.com/bitwarden/clients/issues/8662
- https://github.com/bitwarden/clients/issues/8823
- https://github.com/bitwarden/clients/issues/8841

Amongst others.
